AK/IPv4Address.h

```diff
@@ -76,12 +76,34 @@ class [[gnu::packed]] IPv4Address
         if (string.is_null())
             return {};
         auto parts = string.split_view('.');
-        if (parts.size() != 4)
-            return {};
-        auto a = parts[0].to_uint().value_or(256);
-        auto b = parts[1].to_uint().value_or(256);
-        auto c = parts[2].to_uint().value_or(256);
-        auto d = parts[3].to_uint().value_or(256);
+
+        u32 a;
+        u32 b;
+        u32 c;
+        u32 d;
+
+        if (parts.size() == 1) {
+            a = 0;
+            b = 0;
+            c = 0;
+            d = parts[1].to_uint().value_or(256);
+        } else if (parts.size() == 2) {
+            a = parts[1].to_uint().value_or(256);
+            b = 0;
+            c = 0;
+            d = parts[2].to_uint().value_or(256);
```

Comment:
Have you actually tested it with your case of "1.65793"? I don't see how b and c can end up being set to 1 here.

Reply:
Have a look at the `AK/Tests/` folder, the tests are really easy to write :)

Reply:
That case is a hypothetical extreme case scenario meant to illustrate what my change does **not** support.

Reply:
That case is a hypothetical extreme case scenario meant to illustrate what my change does **not** support.

Reply:
To clarify, my change is meant to deal with one particular scenario, and I tried my best to design it in a way that will let other people to be able to build on top of my change and support cases like `1.65793` later on.